payments-core/src/main/java/com/stripe/android/PaymentSessionViewModel.kt

```diff
@@ -51,6 +52,8 @@ internal class PaymentSessionViewModel(
         savedStateHandle.get<PaymentSessionData>(KEY_PAYMENT_SESSION_DATA)?.let {
             this.paymentSessionData = it
         }
+
+        AnalyticsRequestFactory.regenerateSessionId()
```

Comment:
The top of mind way to fix process death would be to generate/store a UUID in the view model, store it in savedStateHandle, and use it up `set` the session id, rather than have it be automatically regenerated.